core/logger.py

```diff
@@ -46,3 +46,38 @@
 root_logger.addHandler(console)
 
 global_settings.log_file = user_log_dir(log_directory_name)
+
+# manages the logging directory
+# only one directory can have a logger at a time
+directory_logger = None
+
+
+def add_log_to_directory(directory):
+    """
+    Starts up a logging handler that creates a log file in the directory being uploaded
+
+    :param directory: directory to create a logger in
+    :return: None
+    """
+    logging.info("Adding log file to {}".format(directory))
+    log_file = os.path.join(directory, 'irida-uploader.log')
+    global directory_logger
```

Comment:
In theory, if this method was called again with a new `directory` before the call to `remove_directory_logger()` things would get messy (you would have directory loggers still active that don't get properly removed).

Maybe you should add a check to make sure that `directory_logger` is `None` before proceeding (and raise an exception if it is not). You'll then want to reset `directory_logger` to `None` in `remove_directory_logger()`.

Reply:
Good catch,
I have made it so that an exception gets raised (program will exit)

When the multi directory upload is in place the integration tests will verify that loggers are created and removed correctly.